sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -79,247 +99,318 @@ function getProvider(props: any): dynamic.ResourceProvider {
 // This allows the creation of the replacement resource to use the new provider while the deletion of the old
 // resource uses the provider with which it was created.
 
-function cancelRPC(call: any, callback: any): void {
-    callback(undefined, new emptyproto.Empty());
-}
+class ResourceProviderService implements provrpc.IResourceProviderServer {
+    [method: string]: grpc.UntypedHandleCall;
 
-function configureRPC(call: any, callback: any): void {
-    const resp = new provproto.ConfigureResponse();
-    resp.setAcceptsecrets(false);
-    callback(undefined, resp);
-}
-
-async function invokeRPC(call: any, callback: any): Promise<void> {
-    const req: any = call.request;
+    cancel(call: any, callback: any): void {
+        callback(undefined, new emptyproto.Empty());
+    }
 
-    // TODO[pulumi/pulumi#406]: implement this.
-    callback(new Error(`unknown function ${req.getTok()}`), undefined);
-}
+    async configure(
+        call: grpc.ServerUnaryCall<provproto.ConfigureRequest, provproto.ConfigureResponse>,
+        callback: any,
+    ): Promise<void> {
+        const args = call.request.getArgs()!.toJavaScript();
+        const config: Record<string, any> = {};
+        for (const [k, v] of Object.entries(args)) {
+            if (k === providerKey) {
+                continue;
+            }
+            config[k] = rpc.unwrapRpcSecret(v);
+        }
+        this.config = config as any;
```

Comment:
I think we should a. consider adding `private config: ...` to the class definition (apologies if I've missed it) and b. see if we can't do better than `any` here -- could we not do `Record<string, any>`, as expected by `getProvider` and (as best as I tell) what we already have here? Even better would be `Record<string, unknown>` throughout but not sure how feasible that is.